File: Data Structures & Algorithms/majority-element-ii/submission-3.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        mp = Counter(nums)
        ans = []
        # for num in nums:
        #     mp[num] = mp.get(num,0) + 1
        x = len(nums)//3
        for k,v in mp.items():
            if v>x:
                ans.append(k)
        return ans

File: Data Structures & Algorithms/find-duplicate-integer/submission-3.java
class Solution {
    public int findDuplicate(int[] nums) {
        int ans = 0;
        for(int i=0; i<32; i++) {
            int mask = 1<<i;
            int x = 0, y = 0;
            for(int num : nums) {
                if((num & mask) != 0) x++;
            }
            for(int j=1; j<nums.length; j++) {
                if((j & mask) != 0) y++;
            }
            if(x>y)
                ans |= mask;
        }
        return ans;
    }
}
